optuna/visualization/matplotlib/_optimization_history.py

```diff
@@ -94,42 +104,71 @@ def _get_optimization_history_plot(
     cmap = plt.get_cmap("tab10")  # Use tab10 colormap for similar outputs to plotly.
 
     # Prepare data for plotting.
-    trials = [t for t in study.trials if t.state == TrialState.COMPLETE]
+    # trials = [t for t in study.trials if t.state == TrialState.COMPLETE]
+    all_trials = list(
+        itertools.chain.from_iterable(
+            (
+                trial
+                for trial in study.get_trials(deepcopy=False)
+                if trial.state == TrialState.COMPLETE
+            )
+            for study in studies
+        )
+    )
 
-    if len(trials) == 0:
+    if len(all_trials) == 0:
         _logger.warning("Study instance does not contain trials.")
         return ax
 
+    return _get_optimization_histories(studies, target, target_name, ax, cmap)
+    
+def _get_optimization_histories(
+    studies: List[Study],
+    target: Optional[Callable[[FrozenTrial], float]],
+    target_name: str,
+    ax: "Axes",
+    cmap: "Colormap" 
```

Comment:
I think it would be clear to define `cmap` inside this function rather than taking it as an argument of the method.
What do you think about it?